app/controllers/accept_account_shares_controller.rb

```diff
@@ -2,11 +2,16 @@
 
 class AcceptAccountSharesController < ApplicationController
   def show
-    return if user_signed_in?
+    if user_signed_in?
+      return render_notification("You can't accept, because you are the owner") if user_is_owner?
+      return render_notification('Already accepted') if received_share.accepted_at.present?
+
+      return
+    end
 
     session[:after_sign_in_url] = request.fullpath
 
-    shared_email = AccountShare.find_by!(token: ps.fetch(:token)).email
```

Comment:
Can we just reduce the search scope of `AccountShare` here?

Something like that:

```ruby
AccountShare.not_accepted.not_sent_by(current_user).find_by!...
```

The point is that the suggested code has a lot of lines that are not needed. What we need here is just return 404 status (raise active record not found) and that's it.